dvc/render/match.py

```diff
@@ -57,3 +64,125 @@ def match_renderers(
+    from .image_converter import ImageConverter
+    from .vega_converter import VegaConverter
+
+    def_groups = group_defs_plots(data)
+    renderers = []
+
+    renderer_cls = None
+    # TODO integration tests are required to verify that this actually works
+    for plot_id, group in def_groups.items():
+        # TODO how to make sure final_props are there?
+        plot_datapoints = []
+        final_props = {}
+
+        for rev, rev_group in group.items():
+            if (
+                # TODO deal with errored dvc.yaml parsing,
+                # error handler for plots iterator
+                "definitions" in rev_group
+                and "data" in rev_group["definitions"]
+            ):
+                # TODO its flat, create some kind of generator?
+                dvc_file, dvcfile_content = first(
+                    rev_group["definitions"]["data"].items()
+                )
+                plot_def = dvcfile_content.get("data", {})
+                inner_id = first(plot_def)
+
+                (
+                    plot_sources,
+                    plot_properties,
+                ) = extract_config_and_data_sources(
+                    dvc_file, inner_id, plot_def[inner_id]
+                )
+                definitions_data = get_definition_data(data, plot_sources, rev)
+
+                if ImageRenderer.matches(inner_id, None):
+                    renderer_cls = ImageRenderer
+                    if out:
+                        plot_properties["out"] = out
+                    converter = ImageConverter(plot_properties)
+                    for filename, plot_data in definitions_data:
+                        dps, final_props = converter.convert(
+                            revision=rev, filename=filename, data=plot_data
+                        )
+                        plot_datapoints.extend(dps)
+                    renderer_id = inner_id
+
+                else:
+                    # TODO use only the first revision definitions
+                    renderer_cls = VegaRenderer
+                    converter = VegaConverter(plot_properties)
+
+                    skip = []
+                    if len(definitions_data) == 1 and isinstance(
+                        plot_properties.get("y", ""), str
+                    ):
+                        skip.append("generate_y")
+
+                    for filename, plot_data in definitions_data:
+                        dps, final_props = converter.convert(
+                            revision=rev,
+                            filename=filename,
+                            data=plot_data,
+                            skip=skip,
+                            config_path=dvc_file,
+                        )
+                        # TODO test skip
+                        plot_datapoints.extend(dps)
+                    renderer_id = plot_id
+        if "title" not in final_props:
+            final_props["title"] = renderer_id
```

Comment:
Similar to stripping redundant info from the legend, it would be nice to strip `dvc.yaml::` or other path info from plots IDs in the title unless there is a conflict. Alternatively, that info could be shown in a subtitle or some other way to make it less distracting.

Reply:
When an explicit `title:` is set, should we also include the path somewhere, at least when they is a conflict? Right now you can have plots in two different directories with `title: Accuracy` and there is no way to distinguish them in the html.

Reply:
Good point, ill look into that

Reply:
Ok, so in this case, we could relatively simple include that by modyfing `dvc-render`. (It would be backward compatible). I propose to do it after the release.